_posts/2018-03-11-user-testing-warehouse.md

```diff
@@ -0,0 +1,54 @@
+---
+layout: post
+colors:
+  default: 006DAD
+  dark: 005d92
+  light: 669dbd
+
+title: User Testing Warehouse
+summary: Warehouse is the the next generation Python Packaging Index. I recently conducted a series of user tests on the management user interface. This article explores what user testing is, how we applied it and what I'm hoping to do next.
+comments: 'off'
+
+---
+
+User testing is a user experience methodology that places the user at the center of design decisions. UX professionals conduct user tests to challenge their design concepts and validate that the UI works as they intended.
+
+User tests typically involve asking a target user to carry out a series of real world tasks on the UI. Testers can then identify problems in the UI - where the user gets lost or confused, or can't work out how to complete out the assigned task.
+
+Recurring problems are then raised as design issues for the team to fix.
+
+## User Testing Warehouse
+
+It [had always been my intention](http://localhost:4000/2015/12/31/designing-warehouse-an-overview.html) to run user tests on the [Warehouse project](https://pypi.org/), but it wasn't until the project [received MOSS funding](http://pyfound.blogspot.co.uk/2017/11/the-psf-awarded-moss-grant-pypi.html) that I was able to dedicate the amount of time required to set up and conduct these tests.
+
+Having recently completed the pages for logged in users (account settings, manage projects, etc.) these pages seemed like the best candidates for our initial round of testing.
+
+The test I constructed was simple: asking users about their current use of PyPI, then requesting them to complete a series of tasks on the new management pages. For the test, I recruited users who were package owners or maintainers - i.e. people who had experienced the old user interface and who would understand my instructions.
+
+Over three weeks, I ran tests with six different users, with each test taking between 25 minutes and one hour; the duration depending on how much the test subject had to say.
+
+As expected, the tests highlighted a number of problems in the UI, which in turn prompted me to open or tag [20 issues](https://github.com/pypa/warehouse/issues?q=is%3Aissue+label%3A%22user+testing%22) in our issue tracker. The vast majority of these tickets are simple changes that have been tagged as [good first issues](https://github.com/pypa/warehouse/issues?q=is%3Aissue+label%3A%22good+first+issue%22) - which should in turn help with our efforts to outreach to new contributors.
+
+Several of the test participants also raised new ideas, for which I will open new tickets soon.
```

Comment:
All in all, including prep and ticket-opening, how many hours do you believe you will have spent on this round of user testing? That would be an interesting thing to be able to say.